dataset_interfaces/interface.py

```diff
@@ -187,12 +189,56 @@ def load(cls, file_path: Path | str) -> "TestExample":
         d["waits"] = [WaitCreator.unserialise(x) for x in d["waits"]]
         d["example_id"] = file_path.name.removesuffix(".def.json")
         d["dataset_name"] = file_path.parent.name
-        assert d["dataset_name"] in DATASETS_BY_NAME, f"Couldn't find a generator for dataset {d['dataset_name']}."
-        dataset_generator = DATASETS_BY_NAME[d["dataset_name"]]()
         generator_attrs = {"dataset_name", "description", "reset_message"}
         d = {k: d[k] for k in d.keys() if k not in generator_attrs}
         return dataset_generator.create_example(**d)
 
+    def default_waits(self):
+
+        if isinstance(self, DynamicExample):
+            return
+
+        assert len(self.is_question) >= 1, "There are no questions for this test"
+        assert len(self.waits) == 0 or len(self.waits) == len(self.is_question), "Current waits should be empty or the same length as the script"
+        memory_span = self.dataset_generator.memory_span
+
+        # TODO: All this token counting is based on OpenAIs methods, which is not Anthropics - but Aanthopic has no reliable tokeniser released so....
```

Comment:
Not sure. It would be great to have the proper tokeniser being used everywhere. So I will create another ticket for it to investigate more thoroughly.